### ISSUE TITLE
premium features

### ISSUE THREAD

**Issue by collinbarrett** [collinbarrett/FilterLists#3804]
Brainstorming some ideas for premium features for a potential paid feature set down the road. The core information will always be freely accessible, but there are some advanced features I'd love to add that would require revenue to support.

Before any of the features below are started, #372 has to be the priority.

- [ ] unlimited/advanced rule search #33
- [ ] custom alerts: get notified when a rule is added, updated, or removed that contains a custom domain/string
- [ ] FilterLists-hosted mirrors (license-permitting) #464 
  - [ ] bonus: historical versions of lists
- [ ] advanced analytics #371
- [ ] unlimited/enhanced API access

**Comment by iam-py-test** [comment]
How would payment be handled?
>  FilterLists-hosted mirrors (license-permitting)

Paywalling mirrors of public filterlists might get people mad.

**Comment by collinbarrett** [comment]
> How would payment be handled?
TBD. Preferably as privacy-respecting as possible, but honestly it would probably be some 3rd-party like Stripe.

> Paywalling mirrors of public filterlists might get people mad.
Yeah, I agree. I thought maybe there would be some value of hosting mirrors just for historical purposes in case a list ever dies / goes offline. But, I certainly wouldn't suggest that people rely on FilterLists mirrors for daily use. Yeah, I'll probably scratch this idea.

**Comment by iam-py-test** [comment]
No, I think the historical filterlists would be awesome (license abiding), though I personally wouldn't pay for it. I'm just unsure how filterlist maintainers would react.
